app/src/main/java/lt/vilnius/tvarkau/mvp/interactors/ReportPhotoProviderImpl.kt

```
@@ -0,0 +1,18 @@
+package lt.vilnius.tvarkau.mvp.interactors
+
+import android.content.Context
+import lt.vilnius.tvarkau.utils.ImageUtils
+import java.io.File
+
+/**
+ * @author Martynas Jurkus
+ */
+class ReportPhotoProviderImpl(
+        private val context: Context
+) : ReportPhotoProvider {
+
+    override fun convert(file: File): String {
+        return ImageUtils.convertToBase64EncodedString(context, file)
+                ?: throw RuntimeException("Failed to convert photo ${file.path}")
```

Not sure if we should fail silently when impossible to convert photo

---

Can you provide some scenarios when photo conversion will fail in our case ?

---

- OOM when Glide is converting file to bitmap because of huge bounds:
```
        final int requiredWidth = 1600;
        final int requiredHeight = 1600;
```
- file not found for some reason
- because shit happens :)